server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -493,6 +493,8 @@ func (r *statsRecorder) handleTask(ctx context.Context, task *recordStatsTask) {
 	ctx = r.env.GetAuthenticator().AuthContextFromTrustedJWT(ctx, task.invocationInfo.jwt)
 	eg, ctx := errgroup.WithContext(ctx)
 	eg.SetLimit(50) // Max concurrency when copying files from cache->blobstore.
+
+	artifactsUploaded := make(map[string]struct{}, len(task.persist.URIs))
```

Comment:
not sure if we wanna optimize for the common case or the degenerate case here--the bad cases we've seen for this have several thousand outputs with identical values, so perhaps we'd want to avoid pre-allocating capacity in that case?

Reply:
Done.